src/htm/algorithms/SpatialPooler.cpp

```diff
@@ -763,17 +748,7 @@ void applyBoosting_(const UInt i,
 
 
 void SpatialPooler::updateBoostFactorsGlobal_() {
-  Real targetDensity;
-  if (numActiveColumnsPerInhArea_ > 0) {
-    UInt inhibitionArea =
-        (UInt)(pow((Real)(2 * inhibitionRadius_ + 1), (Real)columnDimensions_.size()));
```

Comment:
this was broken for nD SP where SP wasn't a "cube" nxnxn.

Reply:
I don;t see how this is broken.  The volume of an `N` dimensional hypercube with side length `X` is `X ^ N`.

Reply:
Problem is SP's dimensions are not always a hyper-cube, rather a hyper-rectangle, eg. 1000x1x1x1 -> you get area ~ (2*R)^4, compared to ~ 2*R in 1D!